sputnikdao2/src/policy.rs

```diff
@@ -393,6 +404,47 @@ impl Policy {
     }
 }
 
+/// Walks down the policy in JSON format and updates part of it with new value,
+pub(crate) fn update_policy(policy: &Policy, updates: &[PolicyUpdate]) -> Policy {
+    let mut policy_value = near_sdk::serde_json::to_value(policy).expect("Failed to deserialize");
+    for update in updates {
+        let parts: Vec<&str> = update.path.split("/").collect();
+        let mut item = &mut policy_value;
+        let new_value_item =
+            near_sdk::serde_json::from_str::<near_sdk::serde_json::Value>(&update.new_value)
+                .expect("ERR_POLICY_VALUE");
+        // Walk the path to the item to change.
+        for i in 0..parts.len() {
+            let part = parts[i];
+            match item {
+                near_sdk::serde_json::Value::Array(subitems) => {
+                    let index = part
+                        .parse::<usize>()
+                        .expect(&format!("ERR_POLICY_PATH: {} must be integer", part));
+                    if i == parts.len() - 1 {
+                        subitems[index] = new_value_item.clone();
+                    } else {
+                        item = item
+                            .get_mut(index)
+                            .expect(&format!("ERR_POLICY_PATH: {} not found in policy", part));
+                    }
+                }
+                near_sdk::serde_json::Value::Object(subitems) => {
+                    if i == parts.len() - 1 {
+                        subitems.insert(part.to_string(), new_value_item.clone());
```

Comment:
what happens if the `part.to_string()` is a new key and not an existing one? Ex: `skadksdkas` instead of `proposal_bond`